specs/experimental/dispute-game-v2/dispute-game-v2.md

```diff
@@ -0,0 +1,519 @@
+_depth of the claim_. That is, branch claims made higher up in the tree cost more to make.
+
+The calculation of the bond is left to the implementor. It _must_ be a pure function of the definition $B(l)$, where
+$l$ is the length of the branch.
+
+This change introduces a very nice property, which is that the cost of challenging a branch claim is _always_ cheaper
+than creating it, by at least a factor of one scalar unit.
+
+While this is great for a situation when we are dealing with dishonest proposals, this does create a world in
+which the cost of challenging honest proposals is cheaper than creating them. However, in the event of a dishonest
+root claim, it guarantees that the cost to prevent dishonest proposals from going through is less than that of what
+the malicious actor puts on the table.
+
+#### Honest Actor Advantage
+
+Because of the properties of the branch claims as well as trace extension behavior in the `op-challenger`, this
+worst case scenario of "agreement until the last instruction" can be mitigated in the case of challenging a dishonest
+claim in defense of your own. This advantage is also asymmetric - **the malicious actor cannot take advantage of it**,
+as they rely on winning the game via out-spending their opponent and waiting for their clock to run out.
+
+Take the following example:
+
+```mermaid
+flowchart TD
+    A[1] --> B[2]
+    A --> C[3]
+
+    B --> B1[4]
+    B --> B2[5]
+    B1 --> B11[8]
+    B1 --> B12[9]
+    B2 --> B21[10]
+    B2 --> B22[11]
+
+    C --> C1[6]
+    C --> C2[7]
+    C1 --> C11[12]
+    C1 --> C12[13]
+    C2 --> C21[14]
+    C2 --> C22[15]
+
+    style A fill:#A5FE91,color:#000;
+    style B fill:#A5FE91,color:#000;
+    style B1 fill:#A5FE91,color:#000;
+    style B11 fill:#A5FE91,color:#000;
+
+    style C fill:#FF2535,color:#000;
+    style C1 fill:#FF2535,color:#000;
+    style C11 fill:#FF2535,color:#000,stroke-width:3px,stroke-dasharray:3,stroke:#FF1BDF;
+    style C12 stroke-width:3px,stroke-dasharray:3,stroke:#FF1BDF;
+    style C21 stroke-width:3px,stroke-dasharray:3,stroke:#FF1BDF;
+    style C22 stroke-width:3px,stroke-dasharray:3,stroke:#FF1BDF;
+```
+
+Here, assuming trace extension is active in `[12,16)`, the honest actor may short-circuit this dispute immediately,
+without performing further bisection by entering the verifier subcontext with `11` -> `12`. The dishonest actor, on the
+other hand, cannot use the verifier subcontext to win the game, and they are also incentivized to make the honest actor
+expend the most money. Thus, malicious actors seeking to DoS honest proposals do not benefit from this fast-track
+process.
+
+## Bisection Game
+
+The `Bisection Game` protocol is a revisit on the ideas around the initial `FaultDisputeGame` that were never
+implemented. The main goal of this change is to create an abstraction for a layered `IDisputeGame` over ordered data,
+with each layer being unaware of others.
+
+### `ISubContext`
```

Comment:
I think this is the part of the dispute game which could use the most extra thinking. The reason I have this feeling is because SubContexts feel too similar to our current nested claim structure. It would be nice to do more exploration to see if it is possible or reasonable create a dispute game which has a single flat tree but that itself executes the multi-proofs.

@protolambda has had interesting ideas around how we can shape the dispute tree to achieve really interesting results & I think that it might be possible to reuse some of those designs to achieve the same goal as SubContext but without introducing recursive games which resemble our recursive claims

Reply:
Yes, it would be great to discuss in-tree (same bisection game) sub-contexts that can serve the encapsulation of proving the interop state-transition in a series of smaller programs to prove. And there's a parallel here to ZK proof folding schemes that may be interesting to look at.

Reply:
@protolambda has a DB design which seems like it could be used for this. Adding checkpoints at powers of 2 which are easy to compute so that when we bisect the tree it is easy to retrieve all of the information until you get really deep at which point you generate the specific block trace or the specific block ZKP